encoding/unstable_hex.ts

```diff
@@ -0,0 +1,199 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+// This module is browser compatible.
+
+/**
+ * Functions to encode and decode to and from hexadecimal strings.
+ *
+ * ```ts
+ * import { assertEquals } from "@std/assert";
+ * import { encodeHex, type Uint8Array_ } from "@std/encoding/unstable-hex";
+ *
+ * assertEquals(encodeHex("Hello World"), "48656c6c6f20576f726c64");
+ * assertEquals(
+ *   encodeHex(new TextEncoder().encode("Hello World") as Uint8Array_),
+ *   "48656c6c6f20576f726c64",
+ * );
+ * ```
+ *
+ * @see {@link https://www.rfc-editor.org/rfc/rfc4648.html#section-8}
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @module
+ */
+
+import type { Uint8Array_ } from "./_types.ts";
+export type { Uint8Array_ };
+import { calcMax, decode, encode } from "./_common16.ts";
+export { calcMax };
+import { detach } from "./_common_detach.ts";
+
+const alphabet = new TextEncoder().encode("0123456789abcdef");
+const rAlphabet = new Uint8Array(128).fill(16); // alphabet.Hex.length
+alphabet.forEach((byte, i) => rAlphabet[byte] = i);
+new TextEncoder()
+  .encode("ABCDEF")
+  .forEach((byte, i) => rAlphabet[byte] = i + 10);
+
+/**
+ * `encodeHex` takes an input source and encodes it into a hexadecimal string.
+ * If a {@linkcode Uint8Array<ArrayBuffer>} or {@linkcode ArrayBuffer} is
+ * provided, the underlying source will be detached and reused for the encoding.
+ * If you need the input source after providing it to this function, call
+ * `.slice()` to pass in a copy.
+ *
+ * @experimental **UNSTABLE**: New API, yet to be vetted.
+ *
+ * @param input The input source to encode.
+ * @returns The hexadecimal string representation of the input.
+ *
+ * @example Basic Usage
+ * ```ts
+ * import { assertEquals } from "@std/assert";
+ * import { encodeHex, type Uint8Array_ } from "@std/encoding/unstable-hex";
+ *
+ * assertEquals(encodeHex("Hello World"), "48656c6c6f20576f726c64");
+ * assertEquals(
+ *   encodeHex(new TextEncoder().encode("Hello World") as Uint8Array_),
+ *   "48656c6c6f20576f726c64",
+ * );
+ * ```
+ */
+export function encodeHex(
```

Comment:
Feels like there's no need of unstable version of this as this seems completely compatible with the existing version. What do you think?

Reply:
There is a difference in that a Uint8Array or ArrayBuffer past in will detach the buffer to reuse making the user no longer have access to the original content.

Reply:
Ah, ok. That seems significant difference. Ok, let's land this as unstable